javalib/src/main/scala/java/nio/file/Files.scala

```diff
@@ -585,28 +578,34 @@ object Files {
                    currentDepth: Int,
                    options: Array[FileVisitOption],
                    visited: SSet[Path]): SStream[Path] = {
-    start #:: FileHelpers
-      .list(start.toString, (n, t) => (n, t))
-      .toStream
-      .flatMap {
-        case (name, tpe)
-            if tpe == DT_LNK && options.contains(
-              FileVisitOption.FOLLOW_LINKS) =>
-          val path       = start.resolve(name)
-          val newVisited = visited + path
-          val target     = readSymbolicLink(path)
-          if (newVisited.contains(target))
-            throw new FileSystemLoopException(path.toString)
-          else walk(path, maxDepth, currentDepth + 1, options, newVisited)
-        case (name, tpe) if tpe == DT_DIR && currentDepth < maxDepth =>
-          val path = start.resolve(name)
-          val newVisited =
-            if (options.contains(FileVisitOption.FOLLOW_LINKS)) visited + path
-            else visited
-          walk(path, maxDepth, currentDepth + 1, options, newVisited)
-        case (name, _) =>
-          start.resolve(name) #:: SStream.Empty
-      }
+    start #:: {
+      if (!isDirectory(start, linkOptsFromFileVisitOpts(options))) SStream.empty
+      else
```

Comment:
Consider adding braces around that `else`, since it isn't a one-liner:
```suggestion
      else {
```